cpp/src/streaming/cudf/table_chunk.cpp

```diff
@@ -141,14 +159,18 @@ TableChunk TableChunk::spill_to_host(BufferResource* br) {
                 br->move(std::move(packed_columns_->gpu_data), stream_)
             );
         } else {
-            RAPIDSMPF_FAIL("all three data pointers are null");
+            auto packed_columns = cudf::pack(table_view(), stream_, br->device_mr());
+            packed_data = std::make_unique<PackedData>(
+                std::move(packed_columns.metadata),
+                br->move(std::move(packed_columns.gpu_data), stream_)
+            );
         }
     }
+
     // Spill data to host memory.
     auto [res, _] = br->reserve(MemoryType::HOST, packed_data->data->size, false);
     packed_data->data = br->move(std::move(packed_data->data), res);
```

Comment:
I guess not for this PR, but all the above methods calling `cudf::pack` are making allocations with reservations...